tests/cnf/ran/ztp/tests/ztp-argocd-hub-templating.go

```diff
@@ -0,0 +1,218 @@
+		if err == nil {
+			err = secretBuilder.Delete()
+			Expect(err).ToNot(HaveOccurred(), "Failed to delete test secret")
+		}
+	})
+
+	When("full access to ACM hub templating is used", func() {
+		BeforeEach(func() {
+			By("checking the TALM version")
+			versionInRange, err := ranhelper.IsVersionStringInRange(RANConfig.HubOperatorVersions[ranparam.TALM], "4.16", "")
+			Expect(err).ToNot(HaveOccurred(), "Failed to check if TALM version is in range")
+
+			if !versionInRange {
+				Skip("This test requires a TALM version of at least 4.16")
+			}
+		})
+
+		// 54240 - Hub-side ACM templating with TALM
+		It("should allow templating with the printf function", reportxml.ID("54240"), func() {
+			setupHubTemplateTest(tsparams.ZtpTestPathTemplatingPrintf)
+
+			By("validating the policy reaches compliant status")
+			policy, err := ocm.PullPolicy(HubAPIClient, tsparams.HubTemplatingPolicyName, tsparams.TestNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to get policy from hub cluster")
+
+			err = policy.WaitUntilComplianceState(policiesv1.Compliant, tsparams.ArgoCdChangeTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Failed to wait for policy to become compliant")
+		})
+
+		// 54240 - Hub-side ACM templating with TALM
+		It("should allow templating with the prinf function", reportxml.ID("54240"), func() {
+			secretBuilder := secret.NewBuilder(
+				HubAPIClient, RANConfig.Spoke1Name+"-sriovdata", tsparams.TestNamespace, corev1.SecretTypeOpaque)
+			secretBuilder.Definition.StringData = map[string]string{"vlan": "MTEwCg=="}
+			_, err := secretBuilder.Create()
+			Expect(err).ToNot(HaveOccurred(), "Failed to create secret")
+
+			setupHubTemplateTest(tsparams.ZtpTestPathTemplatingFromSecret)
+
+			By("validating the policy reaches compliant status")
+			policy, err := ocm.PullPolicy(HubAPIClient, tsparams.HubTemplatingPolicyName, tsparams.TestNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to get policy from hub cluster")
+
+			err = policy.WaitUntilComplianceState(policiesv1.Compliant, tsparams.ArgoCdChangeTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Failed to wait for policy to become compliant")
+		})
+
+		// 54240 - Hub-side ACM templating with TALM
+		It("should report an error for using autoindent function where not allowed", reportxml.ID("54240"), func() {
+			setupHubTemplateTest(tsparams.ZtpTestPathTemplatingAutoIndent)
+
+			By("validating TALM reported a policy error")
+			assertTalmPodLog(HubAPIClient, "policy has hub template error")
+
+			By("validating the specific error using the policy message")
+			err := helper.WaitForPolicyMessageToContainSubstring(
+				HubAPIClient,
+				tsparams.TestNamespace+"."+tsparams.HubTemplatingPolicyName,
+				RANConfig.Spoke1Name,
+				"wrong type for value; expected string; got int")
+			Expect(err).ToNot(HaveOccurred(), "Failed to validate error using policy message")
+		})
+
+		It("should allow a previously invalid templating function", func() {
```

Comment:
This test case can be removed, and a new valid template should be added to include the new valid configs. 

This should be created with existing content: 
http://registry.kni-qe-0.lab.eng.rdu2.redhat.com:3000/kni-qe/ztp-site-configs/src/workers-4.16/policygentemplates/ztp-test/hub-templating-valid-4.15

And the hub-templating-valid should hold the latest starting 4.16.

When the test cases were created, the idea was negative test cases require one template for each test case in order to validate the proper error message. However, valid templates should be consolidated into single test to reduce execution time. This approach will make it easier to add new valid scenarios. i.e., minimum eco-gotest change is needed. 

This is an example of how different valid test cases are handled between 4.11 and 4.12+: 
https://gitlab.cee.redhat.com/cnf/cnf-gotests/-/blob/release-4.14/test/ran/ztp/tests/ztp_argocd_hub_templating.go?ref_type=heads#L205-212

Sorry I forgot about the history when I review the cnf-gotests PR until I see the big picture in this PR. 

CC @josclark42